extensions/wikia/AdMixExperiment/scripts/noVideo1.js

```diff
@@ -0,0 +1,188 @@
+require(['wikia.throttle'], function (throttle) {
+	$(function () {
+
+		var $adAndRecircWrapper;
+		var $footer = $('#WikiaFooter');
+		var $rail = $('#WikiaRail');
+		var $recirc;
+		var $recircWrapper;
+		var $visibleElementBeforeWrapper;
+		var $window = $(window);
+
+		var adMixRecircWrapperHeight;
+		var bottomMargin = 20;
+		var breakpointSmall = 1023;
+		var globalNavigationHeight = $('#globalNavigation').outerHeight(true);
+		var gapSize = 200;
+		var noRecircAdBottomPosition = 5;
+		var recircEnabled = true;
+		var viewportHeight = $(window).height();
+		var viewportWidth = $(window).width();
+		var visibleElementBeforeWrapperHeight;
+
+		function getStartPosition() {
+			var visibleElementBeforeWrapperBottom = parseInt($visibleElementBeforeWrapper.offset().top, 10) +
+				visibleElementBeforeWrapperHeight;
+
+			if (recircEnabled) {
+				return visibleElementBeforeWrapperBottom - globalNavigationHeight;
+			} else {
+				return visibleElementBeforeWrapperBottom + noRecircAdBottomPosition -
+					(viewportHeight - $adAndRecircWrapper.outerHeight(true));
+			}
+		}
+
+		function getStopPosition() {
+			var stopPoint = parseInt($footer.offset().top, 10);
+			adMixRecircWrapperHeight = $adAndRecircWrapper.outerHeight(true);
+
+			return stopPoint - globalNavigationHeight - adMixRecircWrapperHeight - bottomMargin;
+		}
+
+		function getMidPosition(startPosition) {
+			return startPosition + gapSize;
+		}
+
+		function reset() {
+			if (recircEnabled) {
+				$recircWrapper.css('margin-bottom', gapSize + 'px');
+
+				$recirc.css({
+					position: '',
+					top: '',
+					width: ''
+				});
+			}
+
+			$adAndRecircWrapper.css({
+				bottom: '',
+				position: '',
+				top: '',
+				width: ''
+			});
+		}
+
+		function scrollBeforeMid() {
+			var adAndRecircWrapperStyles = {
+				position: '',
+				width: ''
+			};
+
+			if (recircEnabled) {
+				$recircWrapper.css('margin-bottom', gapSize + 'px');
+
+				$recirc.css({
+					position: 'fixed',
+					top: globalNavigationHeight + 'px',
+					width: '280px'
+				});
+
+				adAndRecircWrapperStyles.top = '';
+			} else {
+				adAndRecircWrapperStyles.bottom = '';
+			}
+
+			$adAndRecircWrapper.css(adAndRecircWrapperStyles);
+		}
+
+		function scrollAfterMid() {
+			var adAndRecircWrapperStyles = {
+				position: 'fixed',
+				width: '300px'
+			};
+
+			if (recircEnabled) {
+				$recircWrapper.css('margin-bottom', '0');
+
+				$recirc.css({
+					position: '',
+					top: '',
+					width: ''
+				});
+
+				adAndRecircWrapperStyles.top = globalNavigationHeight + 'px';
+			} else {
+				adAndRecircWrapperStyles.bottom = noRecircAdBottomPosition + 'px';
+			}
+
+			$adAndRecircWrapper.css(adAndRecircWrapperStyles);
+		}
+
+		function scrollAtBottom(stopPosition) {
+			// FIXME cache it? at least the element
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1134)